docs/spec/qualifiers.rst

```diff
@@ -147,19 +147,27 @@ be initialized in the ``__init__`` method (except in stub files)::
        def __init__(self) -> None:
            self.x = 1  # Good
 
-Type checkers should infer a final attribute that is initialized in
```

Comment:
I think it's worth explicitly addressing the case discussed in https://discuss.python.org/t/treatment-of-final-attributes-in-dataclass-likes/47154/31 by saying that "initialized in the ``__init__`` method" includes implicit initialization in dataclasses. Perhaps on line 141 change "except in stub files" to "except in stub files and dataclasses".

Reply:
I agree, good catch. Pushed an update that makes this more explicit on both sides (the Final docs and the dataclass docs); let me know what you think.